src/integration_tests.zig

```diff
@@ -378,6 +378,34 @@ test "in-place upgrade" {
     cluster.workload_finish();
 }
 
+test "recover smoke" {
+    if (builtin.os.tag != .linux) {
+        return error.SkipZigTest;
+    }
+
+    const replica_count = TmpCluster.replica_count;
+    const seed = std.crypto.random.int(u64);
+    log.info("seed = {}", .{seed});
+
+    var cluster = try TmpCluster.init(.{ .seed = seed });
+    defer cluster.deinit();
+
+    for (0..replica_count) |replica_index| {
+        try cluster.replica_install(replica_index, .past);
+    }
+    try cluster.replica_format(0);
+    try cluster.replica_format(1);
+    try cluster.workload_start(.{ .transfer_count = 200_000 });
+    try cluster.replica_spawn(0);
+    try cluster.replica_spawn(1);
+    std.time.sleep(2 * std.time.ns_per_s);
+
+    try cluster.replica_reformat(2);
+    try cluster.replica_kill(1);
```

Comment:
Hm, why isn't this saying 

```
try cluster.replica_kill(2);
try cluster.replica_reformat(2);
```

I guess we could assert in format&reformat that replicas aren't already running?

Reply:
Ah, wait, we never _start_ replica 2? So this reformat is equivalent to normal format. Why though? Wouldn't it be spicier to let 2 run for a bit, and then kill it?